### DISCUSSION TITLE
withUrqlClient with exchange-graphcache example(next.js)?

### DISCUSSION THREAD

**Discussion by zeing** [urql-graphql/urql#3325] [Q&A] [answered]
I have tried to use exchange-graphcache for pagination with next.js

It isn't worked, I can't find example about it

Anyone have example for this?

```
import { relayPagination } from '@urql/exchange-graphcache/extras'
import { cacheExchange } from '@urql/exchange-graphcache'
import { initUrqlClient, withUrqlClient } from 'next-urql'
```

**Comment by JoviDeCroock** [comment, marked as answer]
Your description is quite unclear here 😅 what didn't work...

Generally how it would look is as followed:

```js
import { fetchExchange } from '@urql/core';
import { relayPagination } from '@urql/exchange-graphcache/extras'
import { cacheExchange } from '@urql/exchange-graphcache'

const cache = cacheExchange({
  resolvers: { YourType: { field: relayPagination() }}
})

export default withUrqlClient(
  ssrExchange => ({
    url: 'https://trygql.formidable.dev/graphql/basic-pokedex',
    exchanges: [cache, ssrExchange, fetchExchange],
  }),
  { ssr: true }
)(YourComponent;
```

**Reply by zeing** [reply]
Sorry for my unclear description.

Just want a full example for next.js with withUrqlClient and exchange-graphcache

But your example codes, it's helpful. 🙏🏽

**Reply by kitten** [reply]
https://github.com/urql-graphql/urql/tree/main/examples

Examples are here, docs have guides too, and there's nothing specific about Next.js + Graphcache 😅

**Reply by zeing** [reply]
Yes , There's not 😢 

Thank you <3

**Reply by kitten** [reply]
Yes, what I mean is: there's nothing specific you have to do to use Next.js and Graphcache together. Hence, there's no example for Next.js+urql+Grapchache

**Reply by zeing** [reply]
I am making infinite pagination in the next.js project 

And I want to cache data when I go next page.

I've decided to use [@urql/exchange-graphcache](https://formidable.com/open-source/urql/docs/api/graphcache/)

Im not sure , It's good way
